nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -71,6 +73,19 @@ in {
       description = "Which package to use for the Nextcloud instance.";
       relatedPackages = [ "nextcloud17" "nextcloud18" "nextcloud19" ];
     };
+    user = mkOption {
+      type = types.str;
+      default = "nextcloud";
+      description = "User of the nextcloud service";
+    };
+    group = mkOption {
```

Comment:
Another possibility here would be to keep running `nextcloud` services (including `php-fpm`) as a non web server user (like the current `nextcloud` user) and instead add the `nextcloud` group to your web servers `extraGroups` option, like so:
```
mkIf cfg.nginx.enable {
  users.users.nginx.extraGroups = [ cfg.group ]; # in other words "nextcloud"
}
```

The reason for this is that the `nginx` user (and group) likely has access to very sensitive files, such as certificates. If `nginx` can read `nextcloud` files (which you want) that is much less of a risk than having `nextcloud` able to read `nginx` files - a serious security concern. This obviously requires that your `nextcloud` files are `g+r`.

Reply:
Sounds like a good idea. The files created by nextcloud seem to be `g+r`, so this shouldn't be an issue.

Reply:
I wonder how to best change the user/group of nextcloud without breaking existing installations. It would require a one time service run as root which changes the ownership. And I would need to have some state file to save that this setup has been completed.
Alternatively I could leave the default user.group to be nginx if `system.stateVersion <= 20.03`, but i don't really like that approach.

Reply:
Forget about it. Since it looks like the user will still be `nextcloud`, only the group needs to be changed from `nginx` to `nextcloud` which is easy to do in the current setup. Also no state needs to be stored.